### PR TITLE
cypress: Add support for watchdog timer update in boot image

### PR DESCRIPTION
**Pull request by romanjoe**
Add watchdog timer to MCUBootApp plus BlinkyApp to implement additional way of boot/upgrade image operation confirmation.

### CODE REVIEW

File: boot/cypress/BlinkyApp/main.c [hunk 1]
```diff
@@ -60,11 +61,14 @@ const cy_stc_gpio_pin_config_t LED_config =
...
 
+#define WATCHDOG_UPD_MESSAGE  "[BlinkyApp] Update watchdog timer started in MCUBootApp to mark successful start of user app\r\n"
+#define WATCHDOG_FREE_MESSAGE "[BlinkyApp] Turn off watchdog timer\r\n"
+
 #ifdef BOOT_IMG
     #define BLINK_PERIOD          (1000u)
     #define GREETING_MESSAGE_VER  "[BlinkyApp] BlinkyApp v1.0 [CM4]\r\n"
     #define GREETING_MESSAGE_INFO "[BlinkyApp] Red led blinks with 1 sec period\r\n"
-#elif UPGRADE_IMG
+#elif defined UPGRADE_IMG
```
Comment (d3zd3z) [inline boot/cypress/BlinkyApp/main.c:71]: In the rest of the code, we conventionally use parens, as in: `#elif defined(UPGRADE_IMG)`?

File: boot/cypress/MCUBootApp/main.c [hunk 2]
```diff
@@ -121,6 +126,12 @@ int main(void)
         if (fih_eq(fih_rc, FIH_SUCCESS))
         {
             BOOT_LOG_INF("User Application validated successfully");
+            /* intitalize watchdog timer. it should be updated from user app
```
Comment (d3zd3z) [inline boot/cypress/MCUBootApp/main.c:129]: Also, stylistically, the rest of the code in MCUboot usually has '*' on each line.

**Review by d3zd3z — APPROVED** [review]
A couple of stylistic suggestions to make the code a little more like the rest of MCUboot, however.

File: boot/cypress/MCUBootApp/main.c (hunk 2 above)
Comment (utzig) [inline boot/cypress/MCUBootApp/main.c:129]: typo: `intitalize` -> `initialize`

**Review by utzig — APPROVED** [review]
LGTM

**Comment by d3zd3z** [comment]
@romanjoe Did you want to fix the minor issues, or merge this as is?

**Pull request merged**